src/Microsoft.Azure.WebJobs.Host/Scale/DefaultHostProcessMonitor.cs

```diff
@@ -0,0 +1,259 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+// Some of the logic in this file was moved from https://github.com/Azure/azure-functions-host/blob/852eed9ceef6ef56b431428a8eb31f1bd9c97f3b/src/WebJobs.Script/Scale/HostPerformanceManager.cs 
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Linq;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+
+namespace Microsoft.Azure.WebJobs.Host.Scale
+{
+    internal class DefaultHostProcessMonitor : IHostProcessMonitor, IDisposable
+    {
+        internal const string CpuLimitName = "CPU";
+        internal const string MemoryLimitName = "Memory";
+
+        private const int MinSampleCount = 5;
+        private readonly long _maxMemoryThresholdBytes;
+        private readonly ProcessMonitor _processMonitor;
+        private readonly List<ProcessMonitor> _childProcessMonitors = new List<ProcessMonitor>();
+        private readonly IOptions<ConcurrencyOptions> _options;
+        private readonly object _syncLock = new object();
+
+        private bool _disposed;
+
+        public DefaultHostProcessMonitor(IOptions<ConcurrencyOptions> options, ProcessMonitor processMonitor = null)
+        {
+            _options = options;
+            _processMonitor = processMonitor ?? new ProcessMonitor(Process.GetCurrentProcess());
+
+            if (_options.Value.MemoryThrottleEnabled)
+            {
+                _maxMemoryThresholdBytes = (long) (_options.Value.TotalAvailableMemoryBytes * (double)_options.Value.MemoryThreshold);
+            }
+        }
+
+        public void Register(Process process)
```

Comment:
Will rename (Unregister below as well). I removed the "for testing" since it caused you confusion. This method is the main entry point, and the other method taking a ProcessMonitor facilitates mock testing (since the tests don't actually spin up a bunch of child processes).